lib/inventoryware/config.rb

```diff
@@ -47,27 +48,53 @@ def respond_to_missing?(s)
       end
     end
 
-    attr_reader :yaml_dir, :templates_dir, :helpers_dir, :req_files,
-                :all_files, :templates_config_path, :plugins_dir, :req_keys
+    attr_reader :yaml_dir, :templates_dir, :templates_config_path, :plugins_dir
 
-    def root_dir
-      @root_dir ||= File.expand_path('../..', __dir__)
+    def initialize
+      @yaml_dir = XDG::Environment.new.data_home.join("flight/inventory")
+      @templates_dir = File.expand_path('../../templates', __dir__)
+      @templates_config_path = File.expand_path('../../etc/templates.yml', __dir__)
+      @plugins_dir = File.expand_path('../../plugins', __dir__)
+
+      FileUtils.mkdir_p yaml_dir
     end
 
-    def initialize
-      @templates_config_path = File.join(root_dir, 'etc/templates.yml')
+    ##
+    # NOTE: The helpers directory contains ruby code for various actions.
+    #       This code can not be easily substituted and seems out of
+    #       place within a "config helpers dir"
+    #
+    #       Consider refactoring within "lib/inventoryware/utils"
+    def helpers_dir
+      @helpers_dir ||= File.expand_path('../../helpers', __dir__)
+    end
```

Comment:
The `helpers` directory contains code which is made available to the template rendering context, and is used only during the rendering of the templates.  That is if a file in `helpers/foo.rb` defines a method `bar()`, any template can run `<%= bar %>`.

This is different to the purpose of the `Utils` module, which is a grab bag of methods which are used internally to flight-inventoryware, not from within a template rendering context.

If/when we support adding additional templates through some kind of plugin/DLC mechanism, it will be important that we also allow the plugin/DLC to ship with its own helpers.

Reply:
:+1: